op-acceptor/cmd/main_test.go

```diff
@@ -151,46 +154,58 @@ func ensureBinaryExists(t *testing.T, projectRoot, binaryPath string) {
 	require.FileExists(t, binaryPath, "op-acceptor binary not found")
 }
 
-func createMockGoMod(t *testing.T, testDir string) {
-	// Initialize the module with go mod init
-	cmd := exec.Command("go", "mod", "init", "test")
-	cmd.Dir = testDir
-	require.NoError(t, cmd.Run(), "Failed to initialize module")
-}
-
 // createMockTest creates a test file that either passes or fails
-func createMockTest(t *testing.T, testDir, packageName, filename string, passing bool) string {
-	// Create package directory
-	packageDir := filepath.Join(testDir, packageName)
-	require.NoError(t, os.MkdirAll(packageDir, 0755))
+func createMockTest(t *testing.T, testDir string, passing bool) string {
+	t.Helper()
 
-	// Create test file
-	testPath := filepath.Join(packageDir, filename)
+	// Create the package directory
+	packageDir := filepath.Join(testDir, passing_dir)
+	if !passing {
+		packageDir = filepath.Join(testDir, failing_dir)
+	}
 
-	// Use a template approach for test content
-	testTemplate := `package %s
+	err := os.MkdirAll(packageDir, 0755)
+	require.NoError(t, err)
 
-import "testing"
+	// Create a go.mod file to make this a valid module
+	goModPath := filepath.Join(packageDir, "go.mod")
+	goModContent := `module test
 
-func %s(t *testing.T) {
-	%s
-}
+go 1.21
```

Comment:
Why not just add a `testdata` directory? It would spare you the trouble of having to dynamically create files

Reply:
I like the flexibility of this approach.
Open to changing it to `testdata`, but I don't have a compelling reason right now.

Reply:
note that for the more  dynamic cases, using `T.TempDir()` as a starting point is probably preferable, as it's unique and cleaned up automatically